### PR TITLE
use `cookie_name/cookie_value` URL for cookie checking API

### PR DESCRIPTION
**Pull request by minrk**
instead of putting the value in the request body, which is verboten for GET requests

/cc @jdfreder

### CODE REVIEW

**Comment by jdfreder** [comment]
Thanks @minrk , this looks good to me.  @jhamrick , should I target this new API in the HubAuth PR for nbgrader?

**Comment by minrk** [comment]
I deprecated the body version, just in case.

File: jupyterhub/singleuser.py [hunk 1]
```diff
@@ -48,10 +48,9 @@ def verify_token(self, cookie_name, encrypted_cookie):
         hub_api_url = self.settings['hub_api_url']
         hub_api_key = self.settings['hub_api_key']
         r = requests.get(url_path_join(
-            hub_api_url, "authorizations/cookie", cookie_name,
```
Comment (ssanderson) [inline jupyterhub/singleuser.py]: Is the text of encrypted_cookie guaranteed to be url-safe and/or does requests automatically escape it intelligently?

Comment (minrk) [inline jupyterhub/singleuser.py]: it's definitely not url-safe, requests does the escaping.

**Comment by minrk** [comment]
This follows the pattern of GitHub, though their tokens aren't cookies, and are url-safe. The other fix is to leave the behavior, and make it a POST request instead of GET, with the cookie value still in the body.

File: jupyterhub/singleuser.py (hunk 1 above)
Comment (minrk) [inline jupyterhub/singleuser.py]: I decided to escape it explicitly. I'm not sure exactly how requests decides what escaping to do, but I'd rather be safe.

**Pull request merged**